controller/service.go

```diff
@@ -174,7 +178,15 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 func (c *controller) clearServiceState(key string, svc *v1.Service) {
 	c.ips.Unassign(key)
 	delete(svc.Annotations, annotationIPAllocateFromPool)
-	svc.Status.LoadBalancer = v1.LoadBalancerStatus{}
+
```

Comment:
MetalLB is still the one in charge of the service, so I think it's fair to erase everything. The hostname makes sense only in association with the ip

Reply:
I see issues in some places. For example when a user requests an ip from a different pool than the one assigned. We would wipe out everything and then if everything goes well, a new ip will be assigned. The hostname will be lost.

We could say that the controller of the user should be able to see this and add the hostname to the newly created ingress. But I don't think we should rely on that. I'm fine if you prefer to simply delete everything. Please let me know.

Reply:
I gave it another thought and I don't think we should go for delete always, especially since we must avoid having two controllers fighting.

Say a user creates an invalid service, such as requesting an ip from a non existing pool, if we delete everything, the user's controller will add the hostname back. This will cause our controller to delete everything again. The cycle will start over. I've tested it.

As you said before, a hostname only makes sense when an IP is allocated. Therefore, I  propose to leave the "Deletion" as it is right now (do not delete ingress with a hostname - clear ip only) and for the "Addition", remove ingresses with hostnames that do not have an ip allocated. For this, I will follow the approach you're suggesting (map + always add IP). This way, we will avoid cycles for erroneous requests and remove invalid hostnames.

What do you think?

Reply:
what I don't like is leaving elements of the ingress slice without ip. I am afraid that this will bite us in the future revealing corner cases that we don't know of today. And in general, add unnecessary complexity to handle hypothetical scenarios that shouldn't be under metallb's scope.

A lb service doesn't work if it doesn't have an exposed ip, so I think we should wipe the loadbalancer.ingress part of the service, which is under metallb's responsability.

If the controller sets the hostname's part of a service that is not exposed (meaning, it doesn't have a lb ip), I think this is a bad behaviour of that controller which wouldn't have any way to resolve that name to an ip that does not exists.